cpp/include/cudf/stream_compaction.hpp

```diff
@@ -231,7 +231,8 @@ enum class duplicate_keep_option {
  *
  * Performance hints:
  * - Always use `cudf::unique` instead of `cudf::distinct` if the input is pre-sorted.
- * - If the input is not pre-sorted, use `cudf::distinct` with a suitable `keep` option.
+ * - If the input is not pre-sorted, use `cudf::distinct` with or without `duplicate_keep_option`
+ *   depending the desired behavior.
```

Comment:
I think this hints are a bit misleading/misguided and treat `cudf::unique` and `cudf::distinct` as two merely two ways of doing the same thing. While it's true that `cudf::unique` can produce the same results as `cudf::distinct` if the input is globally sorted, `cudf::unique` can do more than `cudf::distinct`. Sometimes you don't want a complete deduplication of all values, just deduplication of consecutive equal values.

Reply:
I think we should remove that all. Users should know about this.